include/battle_input.h

```diff
@@ -0,0 +1,211 @@
+#ifndef POKEHEARTGOLD_BATTLE_INPUT_H
+#define POKEHEARTGOLD_BATTLE_INPUT_H
+
+#include "battle_system.h"
+#include "battle_cursor.h"
+#include "battle_finger.h"
+#include "window.h"
+#include "unk_0200E320.h"
+
+typedef struct BattleInputMainScreen {
+    u8 battlerType;
+    u8 battlerId;
+    u8 selectedMon;
+    u8 unk3;
+    s16 hp;
+    u16 maxHp;
+    u8 cancelRun;
+    u8 unused[3];
+} BattleInputMainScreen;
+
+typedef struct BattleInputFightScreen {
+    u16 moveNo[4];
+    u8 pp[4];
+    u8 ppMax[4];
+    u8 battlerType;
+} BattleInputFightScreen;
+
+typedef struct BattleInputTargetScreen {
+    TargetPokemon targetMons[4];
+    u8 battlerType;
+    u8 selectionType;
+} BattleInputTargetScreen;
+
+typedef struct BattleInputYesNoScreen {
+    u16 moveNo;
+    u16 unused;
+} BattleInputYesNoScreen;
+
+typedef union BattleInputScreen {
+    BattleInputMainScreen main;
+    BattleInputFightScreen fight;
+    BattleInputTargetScreen target;
+    BattleInputYesNoScreen yesNo;
+} BattleInputScreen;
+
+typedef struct BattleInput_UnkSub14 {
+    void *unk0;
+    u8 unk4;
+    s8 unk5;
+} BattleInput_UnkSub14;
+
+typedef struct BattleInputMove {
+    u16 moveNo[4];
+    u16 pp[4];
+    u16 ppMax[4];
+} BattleInputMove;
+
+typedef struct BattleInput_UnkSub54_Unk28 {
+    WINDOW window;
+    u16 unkC;
+    u16 unkE;
+} BattleInput_UnkSub54_Unk28;
+
+typedef struct BattleInput_UnkSub54 {
+    BattleInputMove move;
+    u16 *typeIcon[4];
+    BattleInput_UnkSub54_Unk28 unkMove[4];
+    BattleInput_UnkSub54_Unk28 unkPp[4];
+    BattleInput_UnkSub54_Unk28 unkPpMax[4];
+} BattleInput_UnkSub54;
+
+typedef struct BattleInput_UnkSub578_Unk4 {
+    u32 unk0;
+    u32 unk4;
+    u16 type;
+    u16 unkA;
+} BattleInput_UnkSub578_Unk4;
+
+typedef struct BattleInput_UnkSub578 {
+    void *unk0;
+    BattleInput_UnkSub578_Unk4 unk4;
+    u16 unk20;
+} BattleInput_UnkSub578;
+
+typedef struct BattleInput_UnkSub6E0 {
+    s16 xOffset;
+    u8 unk3;
+    u8 state;
+    u8 animationType;
+    u8 delay;
+    u8 count;
+    u8 countMax;
+    u8 loop;
+} BattleInput_UnkSub6E0;
+
+typedef struct BattleInputEffect {
+    s16 state;
+    s16 delay;
+    union {
```

Comment:
is there no better way to do this than a union? I'm guessing not

Reply:
Union is sadly required here